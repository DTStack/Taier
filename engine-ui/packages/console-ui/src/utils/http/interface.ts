
export default interface HttpInterface {
    /**
     * HTTP Get method
     * @param url request URL
     * @param params  request Parameter
     */
    get<R, P = {}>(url: string, params?: P): Promise<R>;
    /**
     * HTTP Post method
     * @param url request URL
     * @param body request body object
     */
    post<R, P = {}>(url: string, body?: P): Promise<R>;
    /**
     * Post an object as a formData object
     * @param url request URL
     * @param params the params object that wait to convert to formData
     */
    postAsFormData<R, P = {}>(url: string, params?: P): Promise<R>;
    /**
     * Post a form element
     * @param url request URL
     * @param form HTML Form element
     */
    postForm<R>(url: string, form: HTMLElement): Promise<R>;
    /**
     * Http request
     * @param url request URL
     * @param options request options
     */
    request<R>(url: string, options?: RequestInit): Promise<R>;
}
