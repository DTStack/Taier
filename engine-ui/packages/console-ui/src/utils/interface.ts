export interface BrowserInter {
    chrome?: string;
    ie?: string;
    edge?: string;
    firefox?: string;
    safari?: string;
    opera?: string;
}
