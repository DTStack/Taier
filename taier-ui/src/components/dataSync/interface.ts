import type { DATA_SOURCE_ENUM } from '@/constant';

/**
 * 前端表单保存的值
 */
export interface ISourceFormField {
	sourceId?: number;
	table?: string | string[];
	/**
	 * Only used in Oracle and PostgreSQL
	 */
	schema?: string;
	where?: string;
	splitPK?: string;
	extralConfig?: string;
	increColumn?: string | number;
	/**
	 * Only used in HDFS
	 */
	path?: string;
	/**
	 * Only used in HDFS
	 */
	fileType?: 'orc' | 'text' | 'parquet';
	/**
	 * Only used in HDFS
	 */
	fieldDelimiter?: string;
	/**
	 * Only used in HDFS
	 */
	encoding?: 'utf-8' | 'gbk';
	/**
	 * Only used in Hive and SparkShrift
	 */
	partition?: string;
}

export interface ITargetFormField {
	sourceId?: number;
	table?: string;
	preSql?: string;
	postSql?: string;
	schema?: string;
	extralConfig?: string;
	partition?: string;
	path?: string;
	fileName?: string;
	fileType?: 'orc' | 'text' | 'parquet';
	fieldDelimiter?: string;
	encoding?: 'utf-8' | 'gbk';
	writeMode?: 'NONCONFLICT' | 'APPEND' | 'insert' | 'replace' | 'update';
}

/**
 * 传给后端的值
 */
export interface ISourceMapProps {
	name?: string;
	sourceId?: number;
	column?: IDataColumnsProps[];
	increColumn?: string | number;
	sourceList?: {
		key: string;
		tables?: string[] | string;
		type: DATA_SOURCE_ENUM;
		name: string;
		sourceId?: number;
	}[];
	extralConfig?: string;
	schema?: string;
	type?: Omit<ISourceFormField, 'sourceId'> & { type?: DATA_SOURCE_ENUM };

	[key: string]: any;
}

export interface ITargetMapProps {
	sourceId?: number;
	name?: string;
	extralConfig?: string;
	column?: IDataColumnsProps[];
	type?: Omit<ITargetFormField, 'sourceId'> & { type?: DATA_SOURCE_ENUM; rowkey?: string };

	[key: string]: any;
}

export interface IKeyMapProps {
	source: IDataColumnsProps[];
	target: IDataColumnsProps[];
}

export interface IDataColumnsProps {
	comment?: string;
	isPart?: boolean;
	key: string | number;
	part?: boolean;
	type: string;
	value?: string;
	index?: string;
	cf?: string;
	format?: string;
}

export interface IChannelFormProps {
	speed: string;
	channel: string;
	record?: number;
	percentage?: number;
	isRestore?: boolean;
	isSaveDirty?: boolean;
	tableName?: string;
	lifeDay?: string | number;
	restoreColumnName?: string | number;
}
