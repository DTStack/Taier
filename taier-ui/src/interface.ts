/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type {
    BINARY_ROW_KEY_FLAG,
    CODE_TYPE,
    DATA_SOURCE_ENUM,
    CREATE_MODEL_TYPE,
    KAFKA_DATA_TYPE,
    MENU_TYPE_ENUM,
    PARAMS_ENUM,
    RESOURCE_TYPE,
    SCHEDULE_DEPENDENCY,
    SCHEDULE_STATUS,
    SOURCE_TIME_TYPE,
    TASK_PERIOD_ENUM,
    TASK_STATUS,
    TASK_TYPE_ENUM,
    DATA_SYNC_MODE,
    UDF_TYPE_VALUES,
    FLINK_VERSIONS,
    DIRTY_DATA_SAVE,
    PythonVersionKind,
} from './constant';

interface IUserProps {}

/**
 * 请求返回体
 */
export interface IResponseBodyProps<T = any> {
    code: number;
    data: T;
    message: string;
    success: boolean;

    [key: string]: any;
}

/**
 * 目录数据结果
 */
export interface CatalogueDataProps {
    id: number;
    /**
     * 目录类型，区分 file 或者 folder 等
     */
    type: string;
    /**
     * 任务类型，非任务为 null
     */
    taskType: TASK_TYPE_ENUM | null;
    resourceType: RESOURCE_TYPE;
    name: string;
    children: CatalogueDataProps[] | null;
    catalogueType: MENU_TYPE_ENUM;
    parentId: number;
    taskDesc?: string;
}

// 运维中心-任务类型，@todo 应该是 Job
export interface ITaskProps {
    gmtModified: number;
    ownerUserId: number;
    ownerUserName: null | string;
    periodType: TASK_PERIOD_ENUM;
    scheduleStatus: SCHEDULE_STATUS;
    taskType: TASK_TYPE_ENUM;
    taskId: number;
    name: string;

    // 工作流任务的子节点
    children?: ITaskProps[];
}

/**
 * 实时任务(Job)管理——任务类型
 */
export interface IStreamJobProps {
    id: number;
    name: string;
    jobId: string;
    status: TASK_STATUS;
    componentVersion: string;
    strategyName: string;
    taskType: TASK_TYPE_ENUM;
    createUserName: string;
    execStartTime: string;
    gmtModified: string;
    modifyUserName: string;
    originSourceType: number;
    createModel: Valueof<typeof CREATE_MODEL_TYPE>;
}

/**
 * 查询任务树的遍历方向
 */
export enum DIRECT_TYPE_ENUM {
    /**
     * 向上遍历
     */
    FATHER = 1,
    /**
     * 向下遍历
     */
    CHILD = 2,
}

/**
 * 任务上下游依赖类型
 */
export interface IUpstreamJobProps {
    taskId: number;
    taskName: string;
    taskType: TASK_TYPE_ENUM;
    scheduleStatus: SCHEDULE_STATUS;
    taskPeriodId: TASK_PERIOD_ENUM;
    gmtCreate: number;
    /**
     * 是否是工作流任务
     */
    isFlowTask: boolean;
    jobId: string;
    tenantId: number;
    tenantName: string;
    /**
     * 计划时间
     */
    cycTime: string;
    operatorId: number;
    /**
     * 操作人名称
     */
    operatorName: string;
    taskGmtCreate: number;
    status: TASK_STATUS;
    parentNode: IUpstreamJobProps[];
    childNode: IUpstreamJobProps[];
}

/**
 * 任务开发-资源类型
 */
export interface IResourceProps {
    createUserId: number;
    gmtCreate: number;
    gmtModified: number;
    id: number;
    modifyUser: null | string;
    modifyUserId: number;
    /**
     * 资源所在的目录 id
     */
    nodePid: number;
    originFileName: string;
    resourceDesc: string;
    resourceName: string;
    resourceType: RESOURCE_TYPE;
    tenantId: number;
    computeType: IComputeType;
    url: string;
}

/**
 * 所有任务类型
 */
export interface IOfflineTaskProps extends ISyncDataProps, IFlinkSQLProps, IWorkflowProps, IFlinkProps, IPythonProps {
    createUserId: number;
    cron: string;
    currentProject: boolean;
    gmtCreate: number;
    gmtModified: number;
    id: number;
    modifyUser: IUserProps | null;
    modifyUserId: number;
    name: string;
    computeType: IComputeType;
    /**
     * 任务绑定的数据源 id
     */
    datasourceId?: number;
    nodePName: string;
    nodePid: number;
    ownerUserId: number;
    scheduleConf: string;
    scheduleStatus: SCHEDULE_STATUS;
    /**
     * 任务配置模式
     */
    createModel: Valueof<typeof CREATE_MODEL_TYPE>;
    /**
     * @deprecated
     * 是否是增量同步模式, 接口要求把该字段放到 sourceMap 中
     */
    syncModel: number;
    sqlText: string;
    taskDesc: string;
    taskParams: string;
    taskPeriodId: TASK_PERIOD_ENUM;
    taskPeriodType: string;
    taskType: TASK_TYPE_ENUM;
    /**
     * 任务依赖
     */
    dependencyTasks: null | ITaskVOProps[];
    taskVariables: null | ITaskVariableProps[];
    tenantId: string | null;
    tenantName: string | null;
    userId: number;
    // 是否支持脏数据记录
    openDirtyDataManage?: boolean;
    // 脏数据管理字段收集
    taskDirtyDataManageVO?: {
        id: number;
        maxRows?: number;
        maxCollectFailedRows?: number;
        outputType?: DIRTY_DATA_SAVE;
        linkInfo?: {
            sourceId: number;
        };
        tableName?: string;
        lifeCycle?: any;
        logPrintInterval?: number;
    };
}

/**
 * Flink 任务类型
 */
interface IFlinkProps {
    mainClass?: string;
    exeArgs?: string;
    resourceIdList?: number[];
}

/**
 * Python 任务类型
 */
interface IPythonProps {
    pythonVersion?: PythonVersionKind;
}

/**
 * 数据同步任务类型
 */
export interface ISyncDataProps {
    settingMap?: IChannelFormProps;
    sourceMap: ISourceMapProps;
    targetMap?: ITargetMapProps;
    taskId: number;
}

export interface IWorkflowProps {
    flowId: number;
    flowName: string;
}

/**
 * 数据同步任务前端表单域类型
 */
export interface ITargetFormField {
    sourceId?: number;
    table?: string;
    preSql?: string;
    postSql?: string;
    schema?: string;
    extralConfig?: string;
    partition?: string;
    path?: string;
    fileName?: string;
    fileType?: 'orc' | 'text' | 'parquet';
    fieldDelimiter?: string;
    encoding?: 'utf-8' | 'gbk';
    writeMode?: 'NONCONFLICT' | 'APPEND' | 'insert' | 'replace' | 'update';
    nullMode?: 'skip' | 'empty';
    writeBufferSize?: number;
    index?: string;
    indexType?: string;
    bulkAction?: number;
}

/**
 * 数据同步任务 TargetMap
 */
export interface ITargetMapProps extends ITargetFormField {
    name?: string;
    column?: IDataColumnsProps[];
    rowkey?: string;
    type?: DATA_SOURCE_ENUM;
}

/**
 * 数据同步任务通道传输类型
 */
export interface IChannelFormProps {
    speed: string;
    channel: string;
    record?: number;
    percentage?: number;
    isRestore?: boolean;
    isSaveDirty?: boolean;
    tableName?: string;
    lifeDay?: string | number;
    restoreColumnName?: string | number;
}

/**
 * 数据同步任务 SourceMap 类型
 */
export interface ISourceMapProps extends ISourceFormField {
    name?: string;
    sourceId?: number;
    /**
     * 数据同步任务「字段映射」
     */
    column?: IDataColumnsProps[];
    sourceList?: {
        key: string;
        tables?: string[] | string;
        type: DATA_SOURCE_ENUM;
        name: string;
        sourceId?: number;
    }[];
    type?: DATA_SOURCE_ENUM;
    /**
     * 同步任务是否增量
     */
    syncModel: DATA_SYNC_MODE;

    [key: string]: any;
}

/**
 * 数据库表字段信息
 */
export interface IDataColumnsProps {
    comment?: string;
    isPart?: boolean;
    /**
     * Generally, it's the unique key for rows
     */
    key: string | number;
    part?: boolean;
    type: string;
    value?: string;
    name?: string;
    /**
     * It's the unique key in FTP for rows
     */
    index?: string;
    cf?: string;
    format?: string;
}

/**
 * 前端表单保存的值
 */
export interface ISourceFormField extends ISourceFieldWithFTP {
    sourceId?: number;
    table?: string | string[];
    /**
     * Only used in Oracle and PostgreSQL
     */
    schema?: string;
    where?: string;
    splitPK?: string;
    extralConfig?: string;
    increColumn?: string | number;
    path?: string;
    /**
     * Only used in HDFS
     */
    fileType?: 'orc' | 'text' | 'parquet';
    /**
     * Only used in HDFS
     */
    fieldDelimiter?: string;
    encoding?: 'utf-8' | 'gbk';
    /**
     * Only used in Hive and SparkShrift
     */
    partition?: string;
    /**
     * 开始行健
     */
    startRowkey?: string;
    /**
     * 结束行健
     */
    endRowkey?: string;
    /**
     * 行健二进制转换
     */
    isBinaryRowkey?: BINARY_ROW_KEY_FLAG;
    /**
     * 每次 RPC 请求获取行数
     */
    scanCacheSize?: number;
    /**
     * 每次 RPC 请求获取列数
     */
    scanBatchSize?: number;
    /**
     * 索引
     */
    index?: string;
    /**
     * 索引类型
     */
    indexType?: string;
    query?: string;
}

export interface ISourceFieldWithFTP {
    path?: string;
    /**
     * It'll rename to fileType when task submitted
     */
    ['fileType|FTP']?: string;
    /**
     * It'll rename to fieldDelimiter when task submitted
     */
    ['fieldDelimiter|FTP']?: string;
    /**
     * 是否包含表头
     */
    isFirstLineHeader?: boolean;
}

/**
 * 离线任务类型中的调度依赖 JSON 字符串格式化后数据类型
 */
export interface IScheduleConfProps {
    /**
     * 初始化的时候该配置为 false
     */
    selfReliance: false | SCHEDULE_DEPENDENCY;
    min?: number | string;
    hour?: number | string;
    /**
     * 数字字符串，如 '0'
     */
    periodType: TASK_PERIOD_ENUM | string;
    /**
     * YYYY-MM-DD 格式
     */
    beginDate: string;
    endDate: string;
    endHour?: string | number;
    endMin?: string | number;
    beginHour?: number | string;
    beginMin?: number | string;
    isFailRetry: boolean;
    gapMin?: string | number;
    /**
     * 数字字符串，如 '5'
     */
    maxRetryNum: number | string;
    isExpire?: boolean;
    isLastInstance?: boolean;
    weekDay?: string | number;
    day?: string | number;
    gapHour?: string | number;
}

/**
 * 任务上下游依赖类型
 */
export type ITaskVOProps = Pick<IOfflineTaskProps, 'id' | 'name' | 'tenantId' | 'tenantName'>;

/**
 * 任务参数
 */
export interface ITaskVariableProps {
    paramCommand: string;
    paramName: string;
    type: PARAMS_ENUM;
}

/**
 * 离线-函数管理属性
 */
export interface IFunctionProps {
    /**
     * 命令格式
     */
    commandFormate: string;
    className?: string;
    createUser?: IUserProps;
    createUserId: number;
    gmtCreate: number;
    gmtModified: number;
    id: number;
    modifyUser?: IUserProps;
    modifyUserId: number;
    name: string;
    nodePid: number;
    paramDesc: string;
    purpose: string;
    sqlText?: string;
    /**
     * 函数类型
     */
    taskType?: TASK_TYPE_ENUM;
    /**
     * UDF类型
     */
    udfType?: UDF_TYPE_VALUES;
    resources?: number;
    type: number;
}

/**
 * 数据源类型
 */
export interface IDataSourceProps {
    dataInfoId: number;
    dataType: string;
    dataTypeCode: DATA_SOURCE_ENUM;
    appNames: string;
    dataDesc: string;
    dataName: string;
    dataVersion: string;
    gmtModified: string;
    isImport: number;
    schemaName: string;
    status: number;
    linkJson: string | null;
}

/**
 * 在同步任务中用到的数据源类型
 */
export interface IDataSourceUsedInSyncProps {
    dataInfoId: number;
    dataName: string;
    dataTypeCode: DATA_SOURCE_ENUM;
    dataType: string;
}

/**
 * flinkSQL 任务的属性
 */
export interface IFlinkSQLProps {
    source: IFlinkSourceProps[];
    sink: IFlinkSinkProps[];
    side: IFlinkSideProps[];
    /**
     * @description 任务类型，目前来说 flinkSQL 暂时只有 1.12
     */
    componentVersion: Valueof<typeof FLINK_VERSIONS>;
}

export interface IFlinkSourceProps {
    charset: CODE_TYPE;
    columns: Partial<{ type: IDataColumnsProps['type']; column: IDataColumnsProps['key'] }>[];
    columnsText: string;
    offset: number;
    offsetReset: string;
    offsetUnit: string;
    offsetValue: string;
    parallelism: number;
    procTime: string;
    schemaInfo: string;
    sourceDataType: KAFKA_DATA_TYPE;
    sourceId: number;
    sourceName: string;
    table: string;
    timeColumn: string;
    timeType: SOURCE_TIME_TYPE;
    timeTypeArr: SOURCE_TIME_TYPE[];
    // 时间戳
    timestampOffset: number;
    timeZone: string;
    topic: string;
    type: DATA_SOURCE_ENUM;
    // 自定义参数
    customParams: { id: string; key?: string; type?: string }[];
}

export interface IFlinkSinkProps {
    collection?: string;
    bucket?: string;
    objectName?: string;
    schema?: string;
    columns: Partial<{ type: IDataColumnsProps['type']; column: IDataColumnsProps['key'] }>[];
    parallelism: number;
    sourceId: number;
    sourceName: string;
    table?: string;
    tableName: string;
    index?: string;
    esId?: string;
    esType?: string;
    rowKey?: string;
    rowKeyType?: string;
    sinkDataType?: KAFKA_DATA_TYPE;
    schemaInfo?: string;
    topic?: string;
    type: DATA_SOURCE_ENUM;
    updateMode: 'append' | 'upsert';
    allReplace?: 'true' | 'false';
    primaryKey?: string | string[];
    bulkFlushMaxActions?: number;
    enableKeyPartitions?: boolean;
    indexDefinition?: string;
    partitionKeys?: string[];
    batchWaitInterval?: number;
    batchSize?: number;
    partitionType?: string;
    columnsText?: string;
    // 自定义参数
    customParams: { id: string; key?: string; type?: string }[];
}

/**
 * FlinkSQL 维表数据结构
 */
export interface IFlinkSideProps {
    type: DATA_SOURCE_ENUM;
    sourceId: number;
    schema?: string;
    table?: string;
    index?: string;
    esType?: string;
    tableName?: string;
    primaryKey?: string[] | string;
    hbasePrimaryKey?: string;
    hbasePrimaryKeyType?: string;
    parallelism?: number;
    columns?: Partial<{
        type: IDataColumnsProps['type'];
        column: IDataColumnsProps['key'];
        targetCol?: string;
    }>[];
    cache?: 'None' | 'LRU' | 'ALL';
    cacheSize?: number;
    cacheTTLMs?: number;
    errorLimit?: number;
    isFaultTolerant?: boolean;
    asyncPoolSize?: number;
    customParams?: { id: string; key?: string; type?: string }[];
}

/**
 * 实时-任务属性参数
 */
export interface IStreamJobParamsProps {
    id: number;
    name: string;
    exeArgs: string;
    sqlText: string;
    taskDesc: string;
    mainClass: string;
    taskParams: string;
    originSourceType: number;
    createModel: number;
    taskType: number;
    targetSourceType: number;
    sourceParams: string;
    sinkParams: string;
    sideParams: string;
    resourceList: IResourceList[];
    additionalResourceList: IResourceList[];
}

/**
 * 实时-资源相关的参数
 */
export interface IResourceList {
    id: number;
    url: string;
    originFileName: string;
    projectId: number;
    resourceDesc: string;
    resourceName: string;
    isAdditionResource: number;
}

/**
 * 当前任务所属计算环境，区分实时任务和离线任务
 */
export enum IComputeType {
    /**
     * 实时任务
     */
    STREAM = 0,
    /**
     * 离线任务
     */
    BATCH = 1,
}

/**
 * 当前任务的任务类型
 */
export enum IJobType {
    /**
     * SQL 类的任务
     */
    SQL = 0,
    /**
     * 非 SQL 类的任务
     */
    OTHERS = 1,
}

export enum RightBarKind {
    /**
     * 任务属性
     */
    TASK = 'task',
    /**
     * 调度依赖
     */
    DEPENDENCY = 'dependency',
    /**
     * 任务参数
     */
    TASK_PARAMS = 'task_params',
    /**
     * 环境参数
     */
    ENV_PARAMS = 'env_params',
    /**
     * 任务设置
     */
    TASK_CONFIG = 'task_config',
    /**
     * 源表
     */
    FLINKSQL_SOURCE = 'flinksql_source',
    /**
     * 结果表
     */
    FLINKSQL_RESULT = 'flinksql_result',
    /**
     * 维表
     */
    FLINKSQL_DIMENSION = 'flinksql_dimension',
    /**
     * 队列管理
     */
    QUEUE = 'queue',
}

export enum DeletedKind {
    notDeleted = 0,
    isDeleted = 1,
}
