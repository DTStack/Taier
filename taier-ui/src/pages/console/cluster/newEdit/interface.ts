import type {
	COMPONENT_TYPE_VALUE,
	ENGINE_SOURCE_TYPE_ENUM,
	FLINK_DEPLOY_NAME,
} from '@/constant';

export interface IComponentProps {
	componentConfig: string;
	componentName: string;
	componentTemplate: string;
	componentTypeCode: number;
	deployType: keyof typeof FLINK_DEPLOY_NAME;
	engineId: ENGINE_SOURCE_TYPE_ENUM;
	gmtCreate: number;
	gmtModified: number;
	versionName: string;
	id: number;
	isDefault: boolean;
	isMetadata?: number;
	storeType: number;
	uploadFileName: string;
	kerberosFileName: string;
	mergeKrb5Content: string;
	paramsFile?: string;
	principal?: string;
	principals?: string;
}

export interface IScheduleComponentComp {
	componentTypeCode: COMPONENT_TYPE_VALUE;
	multiVersion: (IComponentProps | undefined)[];
}

export interface IScheduleComponent {
	components: IScheduleComponentComp[];
	schedulingCode: number;
	schedulingName: string;
}

export interface ISaveCompsData {
	key: number;
	value: string;
}

export interface IModifyComp {
	typeCode: COMPONENT_TYPE_VALUE;
	versionName?: string;
}

export interface IEditClusterRefProps {
	testConnects: (params?: ITestConnectsParams, callBack?: (bool: boolean) => void) => void;
	handleComplete: () => void;
}

export type IConfirmComps = IScheduleComponentComp | IComponentProps | COMPONENT_TYPE_VALUE[];

export interface IGetLoadTemplateParams {
	compVersion?: string;
	storeType?: number;
	deployType?: number;
}

export interface ITestConnectsParams {
	versionName?: string;
	typeCode?: COMPONENT_TYPE_VALUE;
	deployType?: number;
}

export interface IClusterInfo {
	clusterId: string;
	clusterName: string;
}

export interface ITestErrorMsg {
	componentVersion?: string | null;
	errorMsg: string | null;
}

export interface ITestStatusProps {
	clusterResourceDescription?: null | string;
	componentTypeCode: COMPONENT_TYPE_VALUE;
	componentVersion?: string | null;
	multiVersion?: ITestStatusProps[];
	errorMsg: string | null | ITestErrorMsg[];
	result: null | boolean;
}

export interface IVersionProps {
	dependencyKey: string | null;
	dependencyValue: string | null;
	deployTypes: number[] | null;
	id: number;
	key: string;
	required: boolean;
	type: string | null;
	value: string;
	values: IVersionProps[] | null;
}

export type IVersionData = Record<string, IVersionProps[]>;

export interface ICompTemplate {
	dependencyKey?: string;
	dependencyValue?: string;
	key: string;
	required: boolean;
	type: string;
	value: string | string[];
	values?: ICompTemplate[];
	isSameKey?: boolean;
}

export type ISaveComp = (params: Partial<IComponentProps>, type?: string) => void;

export type IHandleConfirm = (action: string, comps: IConfirmComps, mulitple?: boolean) => void;

export type ITestConnects = (params?: ITestConnectsParams, callBack?: (bool: boolean) => void) => void;

export type IHandleCompVersion = (typeCode: number, version: string) => void;
