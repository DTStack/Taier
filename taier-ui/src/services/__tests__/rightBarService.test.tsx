import { RightBarKind } from '@/interface';
import { connect } from '@dtinsight/molecule/esm/react';
import { render } from '@testing-library/react';
import { taskRenderService } from '..';
import RightBarService from '../rightBarService';

jest.mock('../', () => {
    return {
        taskRenderService: {
            renderRightBar: jest.fn(),
        },
    };
});

jest.mock('@/pages/rightBar/flinkDimension', () => {
    return () => <div>flinkDimension</div>;
});
jest.mock('@/pages/rightBar/flinkResult', () => {
    return () => <div>flinkResult</div>;
});
jest.mock('@/pages/rightBar/flinkSource', () => {
    return () => <div>flinkSource</div>;
});
jest.mock('@/pages/rightBar/envParams', () => {
    return () => <div>envParams</div>;
});
jest.mock('@/pages/rightBar/schedulingConfig', () => {
    return () => <div>schedulingConfig</div>;
});
jest.mock('@/pages/rightBar/taskInfo', () => {
    return () => <div>taskInfo</div>;
});
jest.mock('@/pages/rightBar/taskParams', () => {
    return () => <div>taskParams</div>;
});
jest.mock('@/pages/rightBar/taskConfig', () => {
    return () => <div>taskConfig</div>;
});

describe('Test RightBarService', () => {
    afterEach(() => {
        (taskRenderService.renderRightBar as jest.Mock).mockReset();
    });

    it('Should have rightBar text', () => {
        const rightBarService = new RightBarService();
        expect(rightBarService.getTextByKind(RightBarKind.TASK)).toBe('任务属性');
        expect(rightBarService.getTextByKind(RightBarKind.DEPENDENCY)).toBe('调度依赖');
        expect(rightBarService.getTextByKind(RightBarKind.TASK_PARAMS)).toBe('任务参数');
        expect(rightBarService.getTextByKind(RightBarKind.TASK_CONFIG)).toBe('任务配置');
        expect(rightBarService.getTextByKind(RightBarKind.ENV_PARAMS)).toBe('环境参数');
        expect(rightBarService.getTextByKind(RightBarKind.FLINKSQL_SOURCE)).toBe('源表');
        expect(rightBarService.getTextByKind(RightBarKind.FLINKSQL_RESULT)).toBe('结果表');
        expect(rightBarService.getTextByKind(RightBarKind.FLINKSQL_DIMENSION)).toBe('维表');
        expect(rightBarService.getTextByKind(RightBarKind.QUEUE)).toBe('队列管理');
        expect(rightBarService.getTextByKind('')).toBe('未知');
    });

    it('Should have correct component', () => {
        (taskRenderService.renderRightBar as jest.Mock).mockImplementation(() => [
            RightBarKind.TASK,
            RightBarKind.DEPENDENCY,
            RightBarKind.TASK_PARAMS,
            RightBarKind.ENV_PARAMS,
            RightBarKind.TASK_CONFIG,
            RightBarKind.FLINKSQL_SOURCE,
            RightBarKind.FLINKSQL_RESULT,
            RightBarKind.FLINKSQL_DIMENSION,
            '1',
        ]);

        const rightBarService = new RightBarService();

        (connect as jest.Mock)
            .mockReset()
            .mockImplementationOnce((_, Children) => () => <Children current={{ tab: { id: '1' } }} />)
            .mockImplementation((_, Children) => () => <Children current={{ tab: { id: 1 } }} />);

        expect(render(rightBarService.createContent('1')!).asFragment()).toMatchSnapshot();

        expect(render(rightBarService.createContent(RightBarKind.TASK)!).asFragment()).toMatchSnapshot();
        expect(render(rightBarService.createContent(RightBarKind.DEPENDENCY)!).asFragment()).toMatchSnapshot();
        expect(render(rightBarService.createContent(RightBarKind.TASK_PARAMS)!).asFragment()).toMatchSnapshot();
        expect(render(rightBarService.createContent(RightBarKind.ENV_PARAMS)!).asFragment()).toMatchSnapshot();
        expect(render(rightBarService.createContent(RightBarKind.TASK_CONFIG)!).asFragment()).toMatchSnapshot();
        expect(render(rightBarService.createContent(RightBarKind.FLINKSQL_SOURCE)!).asFragment()).toMatchSnapshot();
        expect(render(rightBarService.createContent(RightBarKind.FLINKSQL_RESULT)!).asFragment()).toMatchSnapshot();
        expect(render(rightBarService.createContent(RightBarKind.FLINKSQL_DIMENSION)!).asFragment()).toMatchSnapshot();
    });
});
