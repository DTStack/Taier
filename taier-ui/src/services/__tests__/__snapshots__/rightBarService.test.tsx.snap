// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RightBarService Should have correct component 1`] = `<DocumentFragment />`;

exports[`Test RightBarService Should have correct component 2`] = `
<DocumentFragment>
  <div>
    taskInfo
  </div>
</DocumentFragment>
`;

exports[`Test RightBarService Should have correct component 3`] = `
<DocumentFragment>
  <div>
    schedulingConfig
  </div>
</DocumentFragment>
`;

exports[`Test RightBarService Should have correct component 4`] = `
<DocumentFragment>
  <div>
    taskParams
  </div>
</DocumentFragment>
`;

exports[`Test RightBarService Should have correct component 5`] = `
<DocumentFragment>
  <div>
    envParams
  </div>
</DocumentFragment>
`;

exports[`Test RightBarService Should have correct component 6`] = `
<DocumentFragment>
  <div>
    taskConfig
  </div>
</DocumentFragment>
`;

exports[`Test RightBarService Should have correct component 7`] = `
<DocumentFragment>
  <div>
    flinkSource
  </div>
</DocumentFragment>
`;

exports[`Test RightBarService Should have correct component 8`] = `
<DocumentFragment>
  <div>
    flinkResult
  </div>
</DocumentFragment>
`;

exports[`Test RightBarService Should have correct component 9`] = `
<DocumentFragment>
  <div>
    flinkDimension
  </div>
</DocumentFragment>
`;
